### PR TITLE
chore: upgrade fast-xml-parser

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
* Upgrades fast-xml-parser dependency.
* [fast-xml-parser changelog](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/CHANGELOG.md) for details about how the library has been updated.

### What issues does this PR fix or reference?
* Upgrading this library to pass the Snyk build step in forcedotcom/salesforcedx-vscode

@W-12747980@

### Functionality Before
fast-xml-parser version was outdated

### Functionality After
fast-xml-parser upgraded to latest

### CODE REVIEW

File: src/collections/componentSet.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 /* eslint  @typescript-eslint/unified-signatures:0 */
-import { j2xParser } from 'fast-xml-parser';
+import { XMLBuilder } from 'fast-xml-parser';
```
Comment (gbockus-sf) [inline src/collections/componentSet.ts:8]: worth calling out this was big change from the lib. 
XMLBuilder for js to XML
XMLParser for XML -> js

File: src/collections/componentSet.ts [hunk 2]
```diff
@@ -453,14 +453,15 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
...
-    const j2x = new j2xParser({
+    const builder = new XMLBuilder({
       format: true,
       indentBy: new Array(indentation + 1).join(' '),
       ignoreAttributes: false,
     });
     const toParse = await this.getObject(destructiveType);
     toParse.Package[XML_NS_KEY] = XML_NS_URL;
-    return XML_DECL.concat(j2x.parse(toParse) as string);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```
Comment (gbockus-sf) [inline src/collections/componentSet.ts:463]: is this eslint bypass necessary?

File: src/collections/componentSet.ts [hunk 3]
```diff
@@ -453,14 +453,15 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
    * @param forDestructiveChanges Whether to build a manifest for destructive changes.
    */
   public async getPackageXml(indentation = 4, destructiveType?: DestructiveChangesType): Promise<string> {
-    const j2x = new j2xParser({
+    const builder = new XMLBuilder({
       format: true,
       indentBy: new Array(indentation + 1).join(' '),
```
Comment (mshanemc) [inline src/collections/componentSet.ts:458]: clever, but the +1 feels awkward.

''.padEnd(indentation, ' ')
```suggestion
      indentBy: ''.padEnd(indentation, ' '),

```

File: src/collections/componentSet.ts (hunk 2 above)
Comment (mshanemc) [reply src/collections/componentSet.ts:463]: yes

**Review by mshanemc — APPROVED** [review]
Nice job banging on this until it's right.  Approved with one suggestion

**Review by mshanemc — APPROVED** [review]
Nice job banging on this until it's right.  Approved with one suggestion

**Comment by klewis-sfdc** [comment]
@mshanemc I committed the change to use padEnd(), and some tests failed.  I reverted that change.  Let me know if reverting and keeping it the old way is ok or if you want me to dig further into it to get the usage of padEnd() to work.

**Comment by mshanemc** [comment]
> @mshanemc I committed the change to use padEnd(), and some tests failed. I reverted that change. Let me know if reverting and keeping it the old way is ok or if you want me to dig further into it to get the usage of padEnd() to work.
> 
> 

padEnd is not the cause of that failure, it's changes on main from the ExperieneBundle team that you need to merge into your branch.

**Review by gbockus-sf — APPROVED** [review]
:shipit:

**Pull request merged**